### DISCUSSION TITLE
Streaming APIs for the Filesystem

### DISCUSSION THREAD

**Discussion by wfairclough** [ionic-team/capacitor#3125] [Ideas]
### Feature Request
Streaming APIs in the Filesystem plugin for reading and writing large files on the device

### Platforms
- Android
- iOS
- electron

### Possible Solution
- Browser based solution with [File API](https://www.w3.org/TR/FileAPI) (Seems there is no write ability though)
- A callback solution that provides reading and writing data in chunks
- An event based protocol using a websocket running natively in the background of the app

**Comment by imhoffd** [comment]
We have a few issues for this: https://github.com/ionic-team/capacitor/issues/974 and https://github.com/ionic-team/capacitor/issues/984.

I've personally been meaning to explore the websocket approach (see [my comment here](https://github.com/ionic-team/capacitor/issues/3059#issuecomment-639168635)), but haven't found the time yet.
